js/source/image_source.js

```diff
@@ -57,19 +56,27 @@ ImageSource.prototype = util.inherit(Evented, {
     onAdd: function(map) {
         this.map = map;
         if (this.image) {
-            this.createTile();
+            this.setCoordinates();
         }
```

Comment:
Oh, I am completely with you on why the code exists, it was the intentional or unintentional lack of arguments that got me confused.